platform/darwin/src/MGLLocationManager.h

```diff
@@ -0,0 +1,174 @@
+ does not need to be based on `CLLocationManager`.
+ 
+ */
+@protocol MGLLocationManager <NSObject>
+
+@optional
+
+#pragma mark Determining Location Updates Accuracy
+
+/**
+ Specifies the minimum distance (measured in meters) a device must move horizontally
+ before a location update is generated.
+ 
+ The default value of this property is `kCLDistanceFilterNone` when `MGLMapView` uses  its
+ default location manager.
+ */
+@property(nonatomic, assign) CLLocationDistance distanceFilter;
+
+/**
+ Specifies the accuracy of the location data.
+ 
+ The default value is `kCLLocationAccuracyBest` when `MGLMapView` uses  its
+ default location manager.
+ 
+ @note Determining a location with greater accuracy requires more time and more power.
+ */
+@property (nonatomic, assign) CLLocationAccuracy desiredAccuracy;
+
+/**
+ Specifies the type of user activity associated with the location updates.
+ 
+ The location manager uses this property as a cue to determine when location updates
+ may be automatically paused.
+ 
+ The default value is `CLActivityTypeOther` when `MGLMapView` uses  its
+ default location manager.
+ */
+@property (nonatomic, assign) CLActivityType activityType;
+
+@required
+
+/**
+ The delegate to receive location updates.
+ 
+ Do not set the location manager’s delegate yourself. `MGLMapView` sets this property
+ after the location manager becomes `MGLMapView`’s location manager.
+ */
+@property (nonatomic, weak) id<MGLLocationManagerDelegate> delegate;
+
+#pragma mark Requesting Authorization for Location Services
+
+/**
+ Returns the current localization authorization status.
+ 
+ @see `+[CLLocationManger authorizationStatus]`
+ */
+@property (nonatomic, readonly) CLAuthorizationStatus authorizationStatus;
+
+/**
+ Requests permission to use the location services whenever the app is running.
+ */
+- (void)requestAlwaysAuthorization;
+
+/**
+ Requests permission to use the location services while the app is in
+ the foreground.
+ */
+- (void)requestWhenInUseAuthorization;
+
+#pragma mark Initiating Location Updates
+
+/**
+ Starts the generation of location updates that reports the user's current location.
+ */
+- (void)startUpdatingLocation;
+
+/**
+ Stops the generation of location updates.
+ */
+- (void)stopUpdatingLocation;
+
+#pragma mark Initiating Heading Updates
+
+/**
+ Specifies a physical device orientation.
+ */
+@property (nonatomic) CLDeviceOrientation headingOrientation;
+
+/**
+ Starts the generation of heading updates that reports the user's current hading.
+ */
+- (void)startUpdatingHeading;
+
+/**
+ Stops the generation of heading updates.
+ */
+- (void)stopUpdatingHeading;
+
+/**
+ Dissmisses immediately the heading calibration view from screen.
+ */
+- (void)dismissHeadingCalibrationDisplay;
+
+@end
+
+/**
+ The `MGLLocationManagerDelegate` protocol defines a set of methods that `MGLMapView`
+ uses to receive location updates from the associated location manager.
```

Comment:
This documentation will appear on a separate page from the MGLLocationManager documentation, so it should refer to MGLLocationManager explicitly:

> The `MGLLocationManagerDelegate` protocol defines a set of methods that respond to location updates from an `MGLLocationManager` object that is serving as the location manager of an `MGLMapView`.